File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int globalsum = nums[0];
        int sum = nums[0];
        for(int i=1;i<nums.size();i++){
            if(sum>=0){
                sum += nums[i];
            }
            else
                sum = nums[i];
            globalsum = max(globalsum,sum);
        }
        return globalsum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums){
let max = -Infinity
let temp = 0
for(let i = 0; i<nums.length;i++){
temp = Math.max(nums[i], nums[i] + temp)
max = temp > max ? temp : max
}
return max
};
